Add Plasma Troubleshooing Page (#440)
getsolus/help-center-docs: docs/user/troubleshooting/plasma.md

---
title: Plasma Troubleshooting
summary: Troubleshooting Issues With the Plasma Desktop
---

# Plasma Desktop Troubleshooting

This contains fixes for various issues you might encounter with the Plasma desktop environment.

## Desktop no longer responds to clicks, cursor stuck in "move" mode (cross)

This is often caused by a Firefox bug. It can be triggered by various things, like moving Firefox windows.

> **Symptoms:**

Author: I don't want to make the symptoms headings TOC headings, which is why I didn't use the h3 formatting. Otherwise, in a TOC you would see

Plasma Desktop Troubleshooting
 Desktop no longer responds to clicks, cursor stuck in "move" mode (cross)
  Symptoms
 Application windows look corrupt / incomplete
  Symptoms

etc. "Symptoms" is not valuable as a header item.

- Can't click on anything on the desktop.
- Bringing up windows with keyboard shortcuts (ex: Super for the app menu, F9 for a drop down terminal) works, but they don't get focused. You can't interact with them.
- Mouse cursor is stuck in move mode, it looks like a cross.

> 

Author: I disagree. I think this would just clutter the TOC with needless headings. People can already go to each problem header, which they should read to make sure their situation matches the symptoms. 

Practically, the solutions start just a few lines before the problem summary header, so it's in view anyway.

You may not need to reboot to recover from this state. Killing all firefox processes may be all that's necessary. To do so:

> - Get to a virtual terminal (TTY) with Ctrl-Alt-F3

Reviewer: Use `+` symbols to indicate [keyboard shortcuts](https://developers.google.com/style/ui-elements#keyboard) (Google suggests using the kbd tag, but wrapping everything in inline code blocks should be similar):

```suggestion
1. Open a virtual terminal by pressing `Ctrl+Alt+F3`.
```

Author: Done

> - Run `killall firefox`, this should close all Firefox windows

Reviewer: ```suggestion
2. Execute the following command:
```bash
killall firefox```
```

Author: Done

> - Return to your Plasma session with Ctrl-Alt-F2

Reviewer: ```suggestion
3. Return to the Plasma session by pressing `Ctrl-Alt-F2`. You should now be able to use Plasma normally.
```

Author: Done

> You should now be able to use Plasma normally.

Reviewer: Remove this line.

## Application windows look corrupt / incomplete

**Symptoms**
- Application windows look corrupt. For example, NeoChat not displaying room or user icons.
- Menus in an application are blank.

> You may be able to fix this by clearing KDE / Plasma cache files.

Reviewer: Remove this sentence. More info in the comment I left below.

Author: I'm not sure why it should be removed, even with the reformatting to a multi-step process. To me this is a summary of that process. I reworded it slightly to make that clearer.

> NOTE: You should not be logged into your Plasma session while doing this. It is also highly recommended to back up the folders mentioned below before deleting anything.

Reviewer 2: Maybe convert this to an admonition https://docusaurus.io/docs/2.x/markdown-features/admonitions

Author: Good idea, done.

> NOTE: You should not be logged into your Plasma session while doing this. It is also highly recommended to back up the folders mentioned below before deleting anything.

Reviewer: Since this is part of the procedure, there's no need to add a note about it.

Author: I converted this to a warning. I think it's important to emphasize this because if a user skips steps, thinking they can do this from the regular Plasma session, they need to know it will probably cause them problems. This may also cause us to have questions in the support channels.

> 

Author: Good suggestion, I've reformatted this section accordingly.

> - While still logged into your Plasma session, cd to your home directory, and back up the following directories:

Reviewer: Break this into multiple steps:

```suggestion
1. Open a terminal and go to your home directory.
```bash
cd```

2. Back up the following directories.
```bash
cp -av ~/.kde{,.bak}
cp -av ~/.config{,.bak}
cp -av ~/.local{,.bak}
cp -av ~/share{,.bak}```
```

Author: Done

  ```bash
  cd
  cp -av ~/.kde{,.bak}
  cp -av ~/.config{,.bak}
  cp -av ~/.local{,.bak}
  cp -av ~/share{,.bak}
  ```

> - Log out of Plasma. Boot from a live USB, such as one of the Solus live environments.

Author: Done. Also added a link to the Download page for ease of access.

> - Mount your hard drive's system. For instructions on how to do this see [Mounting your system](https://help.getsol.us/docs/user/troubleshooting/boot-rescue#mounting-your-system)

Reviewer 2: this can be an internal link

Author: Fixed

> - Mount your hard drive's system. For instructions on how to do this see [Mounting your system](https://help.getsol.us/docs/user/troubleshooting/boot-rescue#mounting-your-system)

Reviewer: ```suggestion
5. [Mount your hard drive's file system](./boot-rescue.md#mounting-your-system).
```

Author: Done

> - Change to your user's directory where you mounted your system. In this example, your system has been mounted to `/mount`:

Reviewer: ```suggestion
6. Change to the directory where you mounted your file system. For example:
```

Author: done

  ```bash
  cd /mount/home/username
  ```

> - Run the following from your home directory that you just switched to:

Reviewer: ```suggestion
7. Execute the following commands:
```

Author: Done

  ```bash
  rm -rf .kde
  rm -rf .cache/plasmashell*
  rm -rf .cache/org.kde.dirmodel-qml.kcache
  rm -rf .cache/kioexec/ .cache/krunner/ .cache/ksycoca5*
  rm -rf .cache/krunnerbookmarkrunnerfirefoxdbfile.sqlite

  cd ../.local/
  rm -rf kate/ kded5/ klipper/ knewstuff3/ kscreen/ konsole/ kwalletd/ ksysguard/ kmail2/ kcookiejar/ kactivitymanagerd/

  cd ../share/
  rm -rf dolphin kate kcookiejar kded5 keyrings klipper kmail2 knewstuff3 konsole kscreen ksysguard kwalletd kxmlgui5 plasma_engine_comic plasma plasma_notes org.kde.gwenview

  cd ../.config/
  rm -f plasma*
  rm -rf kde* akonadi* KDE kconf_updaterc baloo* dolphinrc drkonqirc gwenviewrc kmail2rc k*rc katemetainfos
  ```

> Reboot back into your regular Plasma session. Check the application to see if it looks OK now.

Reviewer 2: How do you restore from the `.bak` files if things go sideways?

Author: Good point. I've added steps at the end for how to recover a file or folder. Since this should be rare, I haven't added details as to when you might need to do this. It's only necessary if something does go bork.

> Reboot back into your regular Plasma session. Check the application to see if it looks OK now.

Reviewer: ```suggestion
8. Restart into your regular Plasma session. Verify the issue is fixed.
```

Author: Done

  You should see the files you would usually see in your home directory when logged into Solus normally.
1. Remove the various caches that Plasma uses.
  Execute the following commands:
  ```bash
  rm -rf .kde
  rm -rf .cache/plasmashell*
  rm -rf .cache/org.kde.dirmodel-qml.kcache
  rm -rf .cache/kioexec/ .cache/krunner/ .cache/ksycoca5*
  rm -rf .cache/krunnerbookmarkrunnerfirefoxdbfile.sqlite

  cd ../.local/
  rm -rf kate/ kded5/ klipper/ knewstuff3/ kscreen/ konsole/ kwalletd/ ksysguard/ kmail2/ kcookiejar/ kactivitymanagerd/

  cd ../share/
  rm -rf dolphin kate kcookiejar kded5 keyrings klipper kmail2 knewstuff3 konsole kscreen ksysguard kwalletd kxmlgui5 plasma_engine_comic plasma plasma_notes org.kde.gwenview

  cd ../.config/
  rm -f plasma*
  rm -rf kde* akonadi* KDE kconf_updaterc baloo* dolphinrc drkonqirc gwenviewrc kmail2rc k*rc katemetainfos
  ```
1. Shut down the live environment and remove the live USB drive.
1. Reboot back into your regular Plasma session. Verify the issue is fixed.
1. If you experience system instability, restore the previously backed up folders.

  To restore an entire directory:
  ```
  cp -rav .kde.bak/share/config/ .kde/share/config
  ```

>   After restoring the file(s) or directories, restart your system.

Reviewer: ```suggestion
  After restoring the files or directories, restart your system.
```

---
title: Plasma Troubleshooting
summary: Troubleshooting Issues With the Plasma Desktop
---

# Plasma Desktop Troubleshooting

This contains fixes for various issues you might encounter with the Plasma desktop environment.

## Desktop no longer responds to clicks, cursor stuck in "move" mode (cross)

This is often caused by a Firefox bug. It can be triggered by various things, like moving Firefox windows.

**Symptoms:**
- Can't click on anything on the desktop.
- Bringing up windows with keyboard shortcuts (ex: Super for the app menu, F9 for a drop down terminal) works, but they don't get focused. You can't interact with them.
- Mouse cursor is stuck in move mode, it looks like a cross.

You may not need to reboot to recover from this state. Killing all firefox processes may be all that's necessary. To do so:

1. Open a virtual terminal by pressing `Ctrl+Alt+F3`.
1. Run the following to kill all firefox windows:
  ```bash
  killall firefox
  ```
  This should close all Firefox windows.

> 1. Return to the Plasma session by pressing `Ctrl-Alt-F2`.

Reviewer: ```suggestion
1. Return to the Plasma session by pressing `Ctrl+Alt+F2`.
```

Author: Fixed

  You should now be able to use Plasma normally.

## Application windows look corrupt / incomplete

**Symptoms**
- Application windows look corrupt. For example, NeoChat does not display room or user icons.
- Menus in an application are blank.

> You may be able to fix this by clearing KDE / Plasma cache files, by following the steps below.

Author: I'll update this as you suggest.

I wrote it this way because I have, in the past, had people read docs I wrote that would have not made the connection of the steps below with this paragraph without it being spelled out.

> :::caution

Reviewer: Change this to "important" to comply with the [style guide](https://help.getsol.us/docs/user/contributing/style#admonitions).

```suggestion
:::warning[Important]
```

PS: I know Docusaurus does not have "important", but using Warning with a custom title should do.

Author: Fixed. Note that the syntax should not use brackets

```
:::cautionImportant
```

> You should not be logged into your Plasma session while doing this. It is also highly recommended to back up the folders mentioned below before deleting anything.

Author: As with the other caution admonition, I feel it is worth keeping this in, even if it's part of the steps. Some users may skip steps if they don't realize it's actually important to not be logged into the session, they'll just reboot after. I clarified the instruction a little.

The backup admonition does seem a little redundant tho, so I'll remove that line.

:::

> 1. While still logged into your Plasma session, cd to your home directory:

Reviewer: Don't use commands as verbs. It's confusing for inexperienced users (experienced users won't even need to read the entire step to know what to do)

```suggestion
1. While still logged into your Plasma session, go to your home directory:
```

Author: Fixed

  ```bash
  cd
  ```
1. Run these commands to back up the following directories:
  ```bash
  cp -av ~/.kde{,.bak}
  cp -av ~/.config{,.bak}
  cp -av ~/.local{,.bak}
  cp -av ~/share{,.bak}
  ```

>   The output will show what files were copied. Each directory will have another copy with a .bak extension such as `~/.config.bak`. These can be used to recover configuration files, if necessary.

Author: Fixed

1. Shut down the system.
1. Boot from a live USB, which you can create from one of the [Solus installers](https://getsol.us/download/).
2. [Mount your hard drive's file system](./boot-rescue.md#mounting-your-system).
3. Change to your user's directory where you mounted your file system. For example, if you used `/mount`:
  ```bash
  cd /mount/home/username
  ```
  Replace "username" with the username you use for Solus.
  You can make sure the directory is correct by listing the files:
  ```bash
  ls -laH
  ```
  You should see the files you would usually see in your home directory when logged into Solus normally.

> 1. Now you need to remove the various caches that Plasma uses.

Author: Fixed

  Execute the following commands:
  ```bash
  rm -rf .kde
  rm -rf .cache/plasmashell*
  rm -rf .cache/org.kde.dirmodel-qml.kcache
  rm -rf .cache/kioexec/ .cache/krunner/ .cache/ksycoca5*
  rm -rf .cache/krunnerbookmarkrunnerfirefoxdbfile.sqlite

  cd ../.local/
  rm -rf kate/ kded5/ klipper/ knewstuff3/ kscreen/ konsole/ kwalletd/ ksysguard/ kmail2/ kcookiejar/ kactivitymanagerd/

  cd ../share/
  rm -rf dolphin kate kcookiejar kded5 keyrings klipper kmail2 knewstuff3 konsole kscreen ksysguard kwalletd kxmlgui5 plasma_engine_comic plasma plasma_notes org.kde.gwenview

  cd ../.config/
  rm -f plasma*
  rm -rf kde* akonadi* KDE kconf_updaterc baloo* dolphinrc drkonqirc gwenviewrc kmail2rc k*rc katemetainfos
  ```

> 1. Shut down the live environment and remove the live USB drive.

Reviewer: Does it matter that a lot of steps are marked as "1."? Or Docusaurus will render everything properly?

Author: Docusaurus renders the numbering correctly. You can see this if you check out this branch and run `yarn start`

Since that's the case, it's actually preferable to use 1. for all numbered lines to prevent having to manually number them.

1. Reboot back into your regular Plasma session. Verify the issue is fixed.

> 1. It should not be necessary to recover the cache files that were deleted. If you need to, you can recover a file or directory by copying it from the backup directory to the actual directory.

Author: I like this, I was struggling to come up with a scenario where the user would have to restore the files since I hadn't come across any in personal use or in reading.

  For example, to recover the file `.kde/share/config/kdeglobals`:
  ```
  cp -av .kde.bak/share/config/kdeglobals .kde/share/config/kdeglobals
  ```
  To restore an entire directory:
  ```
  cp -rav .kde.bak/share/config/ .kde/share/config
  ```

>   After restoring the file(s) or directories, log out of Plasma and log back in.

Reviewer: [Avoid using (s), always](https://developers.google.com/style/plurals-parentheses). I personally prefer always defaulting to plural. Also, saying just "log out" or "restart your system" should be enough (it goes without saying they need to log back in to continue their work)

```suggestion
  After restoring the files or directories, restart your system.
```

Author: Fixed
